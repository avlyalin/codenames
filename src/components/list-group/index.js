export { ListGroup } from './list-group';
export { ListGroupItem } from './list-group-item';
