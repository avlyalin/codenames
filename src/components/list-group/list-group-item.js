import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

const ListGroupItem = React.forwardRef(function ListGroupItem(props, ref) {
  const { rounded = true, classes = '', children, ...other } = props;
  return (
    <div
      ref={ref}
      className={classnames(
        'py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t',
        {
          'first:rounded-t-lg': rounded,
          'last:rounded-b-lg': rounded,
          [classes]: classes,
        },
      )}
      {...other}
    >
      {children}
    </div>
  );
});

ListGroupItem.propTypes = {
  rounded: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
  classes: PropTypes.string,
};

export { ListGroupItem };
