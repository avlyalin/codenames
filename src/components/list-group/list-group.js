import React from 'react';
import PropTypes from 'prop-types';

const ListGroup = React.forwardRef(function ListGroup(props, ref) {
  const { children, ...other } = props;
  const items = React.Children.map(children, (child) => {
    return React.cloneElement(child, { ...other });
  });
  return (
    <div ref={ref} className="bg-white">
      {items}
    </div>
  );
});

ListGroup.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};

export { ListGroup };
