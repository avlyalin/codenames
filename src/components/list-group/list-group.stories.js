import React from 'react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import { containerDecorator } from '_storybook/container';
import { ListGroup, ListGroupItem } from '../list-group';

// eslint-disable-next-line import/no-default-export
export default {
  title: 'Components/List group',
  decorators: [containerDecorator(), withKnobs],
};

export const common = () => {
  const rounded = boolean('Rounded', true);
  return (
    <ListGroup rounded={rounded}>
      <ListGroupItem>One</ListGroupItem>
      <ListGroupItem>Two</ListGroupItem>
      <ListGroupItem>Three</ListGroupItem>
    </ListGroup>
  );
};
