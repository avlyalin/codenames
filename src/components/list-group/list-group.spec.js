import React from 'react';
import { render } from '@testing-library/react';
import { ListGroup, ListGroupItem } from '../list-group';

describe('<ListGroup />', () => {
  test('should render list items', () => {
    const { container } = render(
      <ListGroup>
        <ListGroupItem>One</ListGroupItem>
        <ListGroupItem>Two</ListGroupItem>
        <ListGroupItem>Three</ListGroupItem>
      </ListGroup>,
    );
    const listGroup = container.firstChild;

    expect(listGroup).toMatchSnapshot();
  });
  test('should render list items with square borders', () => {
    const { container } = render(
      <ListGroup rounded={false}>
        <ListGroupItem>One</ListGroupItem>
        <ListGroupItem>Two</ListGroupItem>
        <ListGroupItem>Three</ListGroupItem>
      </ListGroup>,
    );
    const listGroup = container.firstChild;

    expect(listGroup).toMatchSnapshot();
  });
});
