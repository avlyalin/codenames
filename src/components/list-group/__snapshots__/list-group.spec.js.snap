// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGroup /> should render list items 1`] = `
<div
  class="bg-white"
>
  <div
    class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t first:rounded-t-lg last:rounded-b-lg"
  >
    One
  </div>
  <div
    class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t first:rounded-t-lg last:rounded-b-lg"
  >
    Two
  </div>
  <div
    class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t first:rounded-t-lg last:rounded-b-lg"
  >
    Three
  </div>
</div>
`;

exports[`<ListGroup /> should render list items with square borders 1`] = `
<div
  class="bg-white"
>
  <div
    class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t"
  >
    One
  </div>
  <div
    class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t"
  >
    Two
  </div>
  <div
    class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t"
  >
    Three
  </div>
</div>
`;
